include/fiction/technology/charge_distribution_surface.hpp

```diff
@@ -317,6 +311,17 @@ class charge_distribution_surface<Lyt, false> : public Lyt
 
         return positions;
     }
+    /**
+     * This function assigns a cell type to a given cell of the underlying cell-level layout.
+     *
+     * @param cell Cell whose type is changed.
+     * @param cell_type Cell type which is assigned to the given cell.
+     */
+    void assign_cell_type(const typename Lyt::cell& cell, const typename Lyt::cell_type& cell_type) noexcept
```

Comment:
Please avoid shadowing:

```suggestion
    void assign_cell_type(const typename Lyt::cell& c, const typename Lyt::cell_type& ct) noexcept
```